types/leaflet/index.d.ts

```diff
@@ -3,1549 +3,1546 @@
 // Definitions by: Alejandro Sánchez <https://github.com/alejo90>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference types="geojson" />
-
-type NativeMouseEvent = MouseEvent;
-type NativeKeyboardEvent = KeyboardEvent;
-
-// Import to avoid conflicts with the GeoJSON class of leaflet
-import GeoJSONFeature = GeoJSON.Feature;
-import GeoJSONLineString = GeoJSON.LineString;
-import GeoJSONMultiLineString = GeoJSON.MultiLineString;
-import GeoJSONPolygon = GeoJSON.Polygon;
-import GeoJSONMultiPolygon = GeoJSON.MultiPolygon;
-import GeoJSONFeatureCollection = GeoJSON.FeatureCollection;
-import GeoJSONGeometryObject = GeoJSON.GeometryObject;
-import GeoJSONGeometryCollection = GeoJSON.GeometryCollection;
-import GeoJSONPoint = GeoJSON.Point;
-import GeoJSONMultiPoint = GeoJSON.MultiPoint;
-import GeoJSONGeoJsonObject = GeoJSON.GeoJsonObject;
-
-declare namespace L {
-    class Class {
-        static extend(props: any): any/* how to return constructor of self extended type ? */;
-        static include(props: any): any /* how to return self extended type ? */;
-        static mergeOptions(props: any): any /* how to return self extended type ? */;
-        static addInitHook(initHookFn: () => void): any/* how to return self extended type ? */;
-    }
+// Renames to avoid conflicts with the GeoJSON class of leaflet
+import {
+    Feature as GeoJSONFeature,
+    LineString as GeoJSONLineString,
+    MultiLineString as GeoJSONMultiLineString,
+    Polygon as GeoJSONPolygon,
+    MultiPolygon as GeoJSONMultiPolygon,
+    FeatureCollection as GeoJSONFeatureCollection,
+    GeometryCollection as GeoJSONGeometryCollection,
+    Point as GeoJSONPoint,
+    MultiPoint as GeoJSONMultiPoint,
+    GeoJsonObject as GeoJSONGeoJsonObject
+} from 'geojson'
+
+export as namespace L
```

Comment:
Annoyingly, it doesn't do that through an UMD wrapper, but it just unconditionally adds its own (ES module) `exports` object to `window.L`. I'm not sure that even compiles with webpack >= 2, and definitely wouldn't work with real modules.